src/main/java/com/minecolonies/coremod/colony/jobs/JobDeliveryman.java

```diff
@@ -63,6 +72,18 @@ private void setupRsDataStore()
                              .getId();
     }
 
+    @Override
+    public void onLevelUp(final int newLevel)
+    {
+        if (getCitizen().getCitizenEntity().isPresent())
+        {
+            final AbstractEntityCitizen worker = getCitizen().getCitizenEntity().get();
+            worker.getAttribute(SharedMonsterAttributes.MOVEMENT_SPEED)
+              .setBaseValue(
+                BASE_MOVEMENT_SPEED + (newLevel > 50 ? 50 : newLevel) * BONUS_SPEED_PER_LEVEL);
+        }
```

Comment:
and when the worker gets fired, we should remove that buff again too.

Reply:
oh good call

Reply:
nvm, exists already on the Dman building. I just moved the speed bonus application to a different place